pkg/services/authn/grpcutils/grpc_authenticator.go

```diff
@@ -75,7 +77,11 @@ func newAuthenticator(auth authn.Authenticator, tracer tracing.Tracer) intercept
 		info, err := auth.Authenticate(ctx, authn.NewGRPCTokenProvider(md))
 		if err != nil {
 			span.RecordError(err)
-			return ctx, err
+			if authn.IsUnauthorizedErr(err) {
+				return nil, status.Error(codes.Unauthenticated, err.Error())
```

Comment:
🔥

Reply:
Unauthorized ?= Unauthenticated

i would think forbidden (eg 401 vs 403)

Reply:
Yeah I realized this is a bit confusing.

The correct http status code for this is 401 Unauthorized https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401. But grpc uses Unauthenticated. 

We should only return 403 when user is authenticated but not authorized to access resource. Maybe I should change the function in authlib to be named `IsUnauthenticatedErr` instead

Reply:
> But grpc uses Unauthenticated.

always need to be a snowflake! 🙃 

Add a comment and then LGTM